staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -1407,3 +1409,193 @@ func isStreamReset(err error) bool {
 	}
 	return false
 }
+
+func TestGetRequestWaitContextWithFake(t *testing.T) {
+	tests := []struct {
+		name                    string
+		defaultRequestWaitLimit time.Duration
+		parent                  func(time.Time) (context.Context, context.CancelFunc)
+		isWatch                 bool
+		reqWaitCtxExpected      bool
+		reqWaitLimitExpected    time.Duration
+	}{
+		{
+			name: "parent context has already expired",
+			parent: func(time.Time) (context.Context, context.CancelFunc) {
+				return &fakeContext{err: context.Canceled}, func() {}
+			},
+			reqWaitCtxExpected: false,
+		},
+		{
+			name:                    "request is a watch",
+			isWatch:                 true,
+			defaultRequestWaitLimit: 10 * time.Second,
+			parent: func(now time.Time) (context.Context, context.CancelFunc) {
+				// the parent deadline should be disregarded for a watch
+				return &fakeContext{deadline: now.Add(2 * time.Minute)}, func() {}
+			},
+			reqWaitCtxExpected:   true,
+			reqWaitLimitExpected: 10 * time.Second, // from now
+		},
+		{
+			name:                    "request is a watch, use received timestamp if available",
```

Comment:
No "if" in this test case, this one definitely has a received timestamp available.

Reply:
In a default server configuration where `WithRequestReceivedTimestamp` is used. In theory it's not necessary required.

Reply:
My point here is about naming the test case.  The idea I get from reading the current version of this PR is that each test case is named after one of several rules or principles of intended behavior that the test case is intended to probe.  But the full truth of the matter is that in general each test case probes the interaction of several rules of intended behavior.  For example, the test case here is also sensitive to rule(s) about whether the given Context is already expired.  So I think the better convention for naming test cases is that each one is named with all the salient characteristics of the test.  For example, this one here might be named something like "watch, context not expired, received timestamp available".